### PR TITLE
Improve setup.py 

### LINKED ISSUE #2: finish setup.py
Finish setup.py script and add install instructions for using it.

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
- Add Makefile target for checking pypi packaging.
- Fix cli installation.
- Improve setup.py/setup.cfg. (Inspired by @ TomasTomecek and @ webknjaz in [ansible-bender](https://github.com/TomasTomecek/ansible-bender/).)

Resolves #2

-----

- [ ] docs update

### CODE REVIEW

File: .git_archival.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+ref-names: $Format:%D$
```
Comment (mpdehaan) [inline .git_archival.txt]: So I've seen a lot of google references to this but I'm unable to find any references online to what this does, can you explain?  Thanks!

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Needed for setuptools-scm-git-archive
+.git_archival.txt  export-subst
```
Comment (mpdehaan) [inline .gitattributes]: I see the comment here but could use a pointer to understanding what this does.  Is there any online doc somewhere?

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+long_description_content_type = text/markdown
+author = Michael DeHaan
+author_email = michael@michaeldehaan.net
+license = Apache License Version 2.0, January 2004
+license_file = LICENSE.md
+classifiers =
+    Development Status :: 4 - Beta
+    Environment :: Console
+    Intended Audience :: Developers
+    License :: OSI Approved :: MIT License
```
Comment (mpdehaan) [inline setup.cfg]: Thanks for filling all of this out.  Note to myself to review the categories.  The license field (MIT) is off at least and I'll probably want to tweak them a little, but this looks quite good.

**Comment by mpdehaan** [comment]
So I appreciate all the work here.

Looking at the upstream repo, it seems to still be in very active development, and I'm not certain how it all works.  There's a bit too much automagic compared to a standard setup.py, and I also don't like how it creates two more dotfiles in the root of the repo.

My general thought here is it is better to have a version file and use setup.py, which I'm still willing to make that change to do.

This is a format that many people are familiar with, and it doesn't introduce an extra setup dependency.  It also means version numbers can be explicit and are not tied to what branches are used.

Finally, I also noticed you work for a software vendor, and I haven't got my CLA integration working yet.  Following precedent also used by that vendor, I've simply added basic CLA terms to CONTRIBUTING.md (which are the same as the Vespene terms), just without the bot integration.  Please look that over and make sure you are ok with those terms.

**Comment by lachmanfrantisek** [comment]
Ok, I can make it in a traditional way. (`version.py`, `setup.py`, ...). Just found  `setuptools-scm-git` very interesting.)

> Finally, I also noticed you work for a software vendor, and I haven't got my CLA integration working yet. 

No problem, this is just my free-time coding..;-)

File: .gitattributes (hunk 2 above)
Comment (lachmanfrantisek) [reply .gitattributes]: Just using the notes in this repo:
https://github.com/Changaco/setuptools_scm_git_archive/

If I understand that correctly, it specifies the version format, using the syntax from `--pretty-format` in `git log`:
```
 ·   %D: ref names without the " (", ")" wrapping.
```

**Comment by mpdehaan** [comment]
Closing this per above comments, I'll get to this soon if you don't want to submit another version, that is totally fine.

setup.py is easy stuff, it's just not really ready for production use just yet so it wasn't an immediate priority.

(It'll should get done this month anyway)

**Pull request closed without merging**